### ISSUE TITLE
Error during template compile of 'CoreModule'

### ISSUE THREAD

**Issue by WellspringCS** [AnthonyNahas/ngx-auth-firebaseui#431]
### Bug Report or Feature Request (mark with an `x`)
ng build -c production
produces compile error

### OS and Version?
Linux Ubuntu 18.04

### Versions
ngx-auth-firebase ui 3.5

Angular CLI: 8.3.0
Node: 10.16.3
OS: linux x64
Angular: 
... 

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.803.0
@angular-devkit/core         8.3.0
@angular-devkit/schematics   8.3.0
@schematics/angular          8.3.0
@schematics/update           0.803.0
rxjs                         6.4.0

### The log given by the failure
ERROR in ../ngx-auth-firebaseui/ngx-auth-firebaseui.ts(141,48): Error during template compile of 'CoreModule'
   Function expressions are not supported in decorators
     Consider changing the function expression into an exported function.

### Mention any other details that might be useful
This bug did not occur in 3.4.2. I discovered the problem after tracking down why the CI/CD deploy failed and discovered it was pulling down 3.5.0 whereas my local copy was 3.4.2

**Comment by AnthonyNahas** [comment]
this is because ur not using the latest version of the library! 

Please clear and clear you npm cache and try again to install the library with the right version

**Comment by WellspringCS** [comment]
Thank you for your note and for this very nice tool, @AnthonyNahas , however to suggest that the problems I was encountering were due to me using the wrong package (and that the solution is for me to fix my improper library references) seem a bit unfair.

When I posted this issue a month ago, 3.5.0 was current (and failing). Reverting to 3.4.2 worked. 

4.0.0 appears to have been released two days ago.

![image](https://user-images.githubusercontent.com/34347780/74741599-ba870d00-5222-11ea-9d86-dd2b262ac652.png)

That said, I'll definitely give 4.x it a go when I get the chance, and thank you for making this OSS tool available.
